source/microbit/mprun.c

```diff
@@ -105,9 +118,11 @@ void mp_run(void) {
     readline_init0();
     microbit_init();
 
-    if (APPENDED_SCRIPT->header[0] == 'M' && APPENDED_SCRIPT->header[1] == 'P') {
+    if (hard_reboot && APPENDED_SCRIPT->header[0] == 'M' && APPENDED_SCRIPT->header[1] == 'P') {
```

Comment:
This changes the behaviour of soft reboot: I got used to it re-executing the appended script and found it useful (sometimes).  Do we want/need to change this behaviour?